compiler/rustc_serialize/src/opaque.rs

```diff
@@ -182,8 +182,8 @@ impl FileEncoder {
             // room to write the input to the buffer.
             unsafe {
                 let src = buf.as_ptr();
-                let dst = MaybeUninit::slice_as_mut_ptr(&mut self.buf).add(buffered);
-                ptr::copy_nonoverlapping(src, dst, buf_len);
+                let dst = self.buf.as_mut_ptr().add(buffered);
+                ptr::copy_nonoverlapping(src, dst.cast(), buf_len);
```

Comment:
Minor nit but could you make casts like this more explicit? When looking at unsafe code I prefer to have as few things to think about as possible.

Reply:
As in `.cast::<u8>()` (and moved up to the `dst` line)? Happy to do that, though I'm curious as to why that's safer. Is it because we're assuming the cast is correct and relying on the type system to complain when pointer types aren't matched up later?

Reply:
Yeah. As I say, it's a minor thing here because the code is fairly straight forward. But in `unsafe` blocks I tend to prefer to keep things very obvious and have as many sanity checks as possible because the nature of it is that the compiler can't always check as much as in safe code. And future refactoring can have a nasty habit of changing things in unexpected ways so I like to make sure the humans and the compiler all clearly agree on what's happening even the human is just looking at a few lines of a diff.

In short, maybe I'm a bit paranoid but I think there's no harm in being too cautious.

Reply:
No no, that makes sense, thanks!